react_app/src/components/ProfileButton.js

```diff
@@ -54,12 +39,24 @@ class ProfileButton extends React.Component {
       }
     } else {
       return (
-        <Button color="inherit" onClick={this.doLogOut}>
+        <Button color="inherit" onClick={() => this.props.logout()}>
```

Comment:
This pattern seems counter intuitive - surely it would be better to just dispatch(logout())?

Reply:
you could do that, but it'll make this component more difficult to test and couples the component itself more closely to redux. This is very much normal (it's what `mapDispatchToProps` exists for)

Reply:
feel free to do it though - many people do. Because this component is `connect`ed, dispatch will be in the props. You can therefore remove `mapDispatchToProps`, and replace the line above with;
```
        <Button color="inherit" onClick={() => this.props.dispatch(logout())}>
```

Reply:
Sorry I am aware that a lot of this is just probably going to sound like soundbytes - happy to explain further if required.

Reply:
I've merged this, and now I'm going to spend a while playing with it, having the action come in via the props seems a little weird but this entire JS learning experience has been like that at first.